### DISCUSSION TITLE
Any method to debug python code call by pybind11 from c++?

### DISCUSSION THREAD

**Discussion by zeroxer** [microsoft/debugpy#1258] [Q&A] [answered]
- I use pybind11 to embed python in my app, and use it to call python code. So there is no python.exe, just my app.exe and the python script hello.py. Dose debugpy support debug this hello.py?
- python module was defined with `PYBIND11_EMBEDDED_MODULE`

**Comment by int19h** [comment, marked as answer]
It does, but you won't be able to use the launch (F5) workflow. You'll need to attach the client to your process using attach-to-PID or attach-to-port. If you have control over the Python code that runs in your host process and can import debugpy and use its API there to set up the debug server, attach-to-port will generally work more reliably:
https://code.visualstudio.com/docs/python/debugging#_debugging-by-attaching-over-a-network-connection
